@navikt/core/react/src/form/file-upload/FileUpload.tsx

```diff
@@ -124,86 +123,42 @@ export const FileUpload = forwardRef<HTMLDivElement, FileUploadProps>(
       }
     }
 
-    if (variant === "button") {
-      return (
-        <div className={className} onDragOver={onDragOver} onDragLeave={onDragLeave} ref={ref}>
-          <label className={cl(buttonClassNames, { "navds-fileuploadbox--dragover": isDraggingOver })}>
-            <UploadIcon fontSize="1.5rem" focusable={false} aria-hidden={true} />
-            <BodyShort as="span">{label}</BodyShort>
-            <Input
-              inputId={inputId}
-              onChange={handleUpload}
-              accept={accept}
-              ariaDescribedby={ariaDescribedby}
-              multiple={multiple}
-            />
-          </label>
-          <Error error={error} errorId={errorId} />
-        </div>
-      )
-    }
+    const isBoxVariant = variant === "box"
 
     return (
-      <div className={className} onDragOver={onDragOver} onDragLeave={onDragLeave} ref={ref}>
-        <label className={cl('navds-fileuploadbox', { "navds-fileuploadbox--error": !!error, "navds-fileuploadbox--dragover": isDraggingOver })}>
-          <BodyShort className="navds-fileuploadbox__text">Dra og slipp</BodyShort>
-          <BodyShort className="navds-fileuploadbox__text">eller</BodyShort>
-          <div className={buttonClassNames}>
-            <UploadIcon fontSize="1.5rem" focusable={false} aria-hidden={true} className="navds-fileuploadbox__icon" />
+      <div className={cl('navds-fileupload', className)} onDragOver={onDragOver} onDragLeave={onDragLeave} ref={ref}>
+        <label className={
+          cl({
+            'navds-fileuploadbox': isBoxVariant,
+            "navds-fileuploadbox--error": !!error && isBoxVariant,
+            "navds-fileupload--dragover": isDraggingOver
+          })
+        }>
+          {isBoxVariant && (<>
```

Comment:
Jeg la til en commit https://github.com/navikt/aksel/pull/2504/commits/4376843ded7f5782a855e321bcba880f98fb355a hvor jeg endret fra å ha én if og tilhørende return per variant og heller kjøre én felles return og gjøre flere if'er inni den returnen for å rendre basert på variant.

Fordelen er at man kan gjenbruke litt mer kode mellom variantene, men ulempen er at det kan være vanskeligere å lese når koden til variantene står om hverandre.

Hva slags pattern foretrekker dere i Aksel?

Reply:
Kommer litt an på kompleksiteten på hva if-testen returnerer. Hvis jeg skjønte variant-`box` riktig så er det dropzone området? Om så kan det bli ganske komplekst etterhvert med events hvis vi ønsker forskjellige visninger basert på hover/drag-over state, så kan kanskje trekkes ut til en egen fil 🤔

Reply:
`box` er dropzone-ja. Den eneste forskjellen er at i `box` vises det to tekster til ("Dra og slipp", "eller") og at man legger på en klasse ekstra som gjør at labelet får padding, border og backgroud-color som styler dropzone-området.

Per nå er det ikke så veldig mye logikk basert på variant, så det er ganske deilig å slippe å duplisere kode eller dra alt ut i ekstra sub-komponenter imo.

Vi kan kanskje ta opp diskusjonen igjen når all funksjonaliteten er ferdig og vi vet hva kompleksiteten blir? 🤔 

Jeg har forresten laget en TODO-liste i PR-beskrivelsen, har dere noen innspill der? Dere kan godt bare redigere beskrivelsen direkte hvis dere har rettigheter til det 🙌